src/List.js

```diff
@@ -1,119 +1,154 @@
-import { useState } from "react";
-import AddItem from "./AddItem";
-import Alert from "./Alert";
-import CompletionButtons from "./CompletionButtons";
-import ListItem from "./ListItem";
+import firebase from 'firebase/compat';
+import { useEffect, useState } from 'react';
+import { generateUniqueID } from 'web-vitals/dist/modules/lib/generateUniqueID';
+
+import AddItem from './AddItem';
+import Alert from './Alert';
+import CompletionButtons from './CompletionButtons';
+import ListItem from './ListItem';
+import SortSelect from './SortSelect';
 
 function List(props) {
-  const [data, setData] = useState(props.listItems);
-  const [idCounter, setIdCounter] = useState(data.length);
-  const [isChecked, setIsChecked] = useState([]);
+  const [numChecked, setNumChecked] = useState(0);
+  const [totalTasks, setTotalTasks] = useState(0);
+  const [editingText, setEditingText] = useState('');
   const [isEditingId, setIsEditingId] = useState(null);
-  const [editingText, setEditingText] = useState("");
   const [showAlert, setShowAlert] = useState(false);
   const [showingAllTasks, setShowingAllTasks] = useState(true);
-
-  function handleAdd(e) {
-    console.log("adding");
-    setData([...data, {
-      id: idCounter,
-      task: ""
-    }]);
-    setIsEditingId(idCounter);
-    setEditingText("");
-    setIdCounter(idCounter + 1);
+  const [newestItem, setNewestItem] = useState(null);
+
+  useEffect(() => {
```

Comment:
Shouldn't be in a useEffect.

You have *all* the tasks in-memory. Just use filter if you need to count how many are checked.